pkg/emptydisk/emptydisk.go

```diff
@@ -0,0 +1,38 @@
+package emptydisk
+
+import (
+	"os"
+	"os/exec"
+	"path"
+	"strconv"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+)
+
+var EmptyDiskBaseDir = "/var/run/libvirt/emptyDisks/"
+
+func CreateTemporaryDisks(vm *v1.VirtualMachine) error {
+
+	for _, volume := range vm.Spec.Volumes {
+
+		if volume.EmptyDisk != nil {
+			// qemu-img takes the size in bytes or kin Kibibytes/Mebibytes/...; lets take bytes
+			size := strconv.FormatInt(volume.EmptyDisk.Capacity.ToDec().ScaledValue(0), 10)
+			file := FilePathForVolumeName(volume.Name)
+			if err := os.MkdirAll(EmptyDiskBaseDir, 0777); err != nil {
+				return err
+			}
+			if _, err := os.Stat(file); os.IsNotExist(err) {
+				if err := exec.Command("qemu-img", "create", "-f", "qcow2", file, size).Run(); err != nil {
```

Comment:
Are there any performance implications to always using qcow2?

For example, does a qcow2 disk that's expanded to it's maximum size perform similar to a raw disk?  If not it's possible we'd want to have the option to use fallocate to create a raw disk in some situations.

Reply:
There are no performance impacts depending on the size when we create the file.

Reply:
Oh I misread your comments. So there is a rw performance impact when using the sparse file. Depending on the test scenarios I found comparisons whish show a slower qcow2 disk performance between 5 and 50%. However it seems like this seems to be mostly related to the sparse file allocation. It is possible to create qcow2 files which are not sparse. That leads to pretty much equal performance like with a raw disk, but then the creation takes much longer.